### PR TITLE
Starting docs homepage updates

### PR DESCRIPTION
**Pull request by dbanksdesign**
*Issue #, if available:*

*Description of changes:*

https://user-images.githubusercontent.com/321279/142049024-1c398ad9-b28b-4215-98cc-a2ca28b46441.mp4

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   },
   "dependencies": {
     "@aws-amplify/ui-react-v1": "npm:@aws-amplify/ui-react",
-    "@codesandbox/sandpack-react": "^0.1.15",
+    "@codesandbox/sandpack-react": "0.1.9",
```
Comment (reesscot) [inline docs/package.json:13]: Just curious, was there something broken in 0.1.15? This is where I wish it was possible to put comments in JSON.

File: docs/src/components/Fragment.tsx [hunk 2]
```diff
@@ -58,35 +47,28 @@ export const Fragment = ({ children, platforms }: FragmentProps) => {
...
+              {error.message.includes('Cannot find module') ? (
+                <>Content missing for {platform}.</>
+              ) : (
+                error.message
+              )}{' '}
+              <Link
+                isExternal
+                href="https://github.com/aws-amplify/amplify-ui/issues/new/choose"
+              >
+                Please open an issue.
```
Comment (reesscot) [inline docs/src/components/Fragment.tsx:60]: Side question - What are we going to show for unsupported platforms for the primitives? Not sure this is the most user friendly message to show for say the Text Field primitive if they click the Vue button.

File: docs/src/components/Layout/Footer.tsx [hunk 3]
```diff
@@ -1,27 +1,71 @@
...
-          policy.
-        </Text>
-        <Link
+    <Flex as="footer" direction="column" className="docs-footer">
+      <Flex direction="row" justifyContent="center">
+        <Logo />
+      </Flex>
+      <View textAlign="center">
+        <Button
+          as="a"
```
Comment (reesscot) [inline docs/src/components/Layout/Footer.tsx:24]: It's pretty awesome how well the polymorphism seems to be working! Seems pretty straightforward to get a button-like link here.

File: docs/src/components/Layout/Footer.tsx [hunk 4]
```diff
@@ -1,27 +1,71 @@
...
+            pathData={icons.DISCORD}
           />
-        </Link>
+          Discord
+        </Button>
+      </View>
+      <Flex direction="row">
+        <View>
+          Amplify open source, documentation and community are supported by
+          Amazon Web Services © {new Date().getFullYear()}, Amazon Web Services,
```
Comment (reesscot) [inline docs/src/components/Layout/Footer.tsx:53]: Thank you for not making us update the copyright year every December 🤣

**Review by reesscot — APPROVED** [review]
LGTM! 🚢

File: docs/package.json (hunk 1 above)
Comment (dbanksdesign) [reply docs/package.json:13]: Yes, it just flat out wouldn't work. See https://github.com/codesandbox/sandpack/issues/135

File: docs/src/components/Layout/Footer.tsx (hunk 3 above)
Comment (dbanksdesign) [reply docs/src/components/Layout/Footer.tsx:24]: Yup!

File: docs/src/components/Layout/Footer.tsx (hunk 4 above)
Comment (dbanksdesign) [reply docs/src/components/Layout/Footer.tsx:53]: lol

File: docs/src/components/Fragment.tsx (hunk 2 above)
Comment (dbanksdesign) [reply docs/src/components/Fragment.tsx:60]: Yea we should definitely update this to be more friendly and be more explicit in the docs how we talk about frameworks.

File: docs/package.json (hunk 1 above)
Comment (dbanksdesign) [reply docs/package.json:13]: Looks like it was fixed in an update, will update the version

Comment (dbanksdesign) [reply docs/package.json:13]: Never mind, I tried it out and it didn't work :(

Comment (ericclemmons) [reply docs/package.json:13]: **EDIT** Just saw that you've re-introduced Sandpack. Thank you!!!

---

@dbanksdesign We can remove this entirely – it was only used for the prototype & would need to be reimplemented once we adopt it again.

I see references to it in:

- `docs/package.json`
- `docs/src/pages/index.page.tsx`
- `docs/src/styles/index.scss`

File: docs/src/components/Fragment.tsx (hunk 2 above)
Comment (ericclemmons) [reply docs/src/components/Fragment.tsx:60]: I would like there to be a Discussion that we can link to and request customers vote & comment on that issue for support.

File: docs/src/pages/index.page.tsx [hunk 5]
```diff
@@ -1,64 +1,104 @@
 import * as React from 'react';
 import { useRouter } from 'next/router';
-// import { Sandpack, codesandboxDarkTheme } from '@codesandbox/sandpack-react';
+import { Sandpack, codesandboxDarkTheme } from '@codesandbox/sandpack-react';
```
Comment (ericclemmons) [inline docs/src/pages/index.page.tsx:3]: Wait, are we actually using this? 😲

File: docs/src/pages/index.page.tsx [hunk 6]
```diff
@@ -117,11 +165,15 @@ const HomePage = ({ colorMode }) => {
...
+        <Heading level={2} textAlign="center" margin={`${tokens.space.xl}`}>
+          Take it for a test drive
+        </Heading>
+        <Card style={{ width: '100%', padding: 0 }} variation="outlined">
+          <Sandpack
             template="react"
             files={{
-              '/App.js': code,
+              '/App.js': { code: code, active: true },
+              '/theme.js': { code: themeCode },
```
Comment (ericclemmons) [inline docs/src/pages/index.page.tsx:176]: Heck yeah!! It's back!!!

File: docs/src/pages/index.page.tsx [hunk 7]
```diff
@@ -133,69 +185,36 @@ const HomePage = ({ colorMode }) => {
             }}
             customSetup={{
               dependencies: {
-                '@aws-amplify/ui-react': '0.0.0-next-202192618169',
+                '@aws-amplify/ui-react': '0.0.0-next-5257a31-20211011212613',
```
Comment (ericclemmons) [inline docs/src/pages/index.page.tsx:188]: I didn't see CodeSandbox UI's supporting `next` tags.  Does it work here?
```suggestion
                '@aws-amplify/ui-react': 'next',
```

**Review by ericclemmons — APPROVED** [review]

File: docs/src/pages/index.page.tsx (hunk 6 above)
Comment (dbanksdesign) [reply docs/src/pages/index.page.tsx:176]: It's ok to have it in, right?

Comment (ericclemmons) [reply docs/src/pages/index.page.tsx:176]: Oh, heck yeah!  I just haven't had the time to revisit it. 

Originally, I had React, Angular, & Vue templates and was feeding it code _directly from our proof-of-concept_.

(This was before we had environments, e2e tests, primitives, etc.)

So let's keep it in! This is what we want long-term.

**Pull request merged**